pkg/logs/internal/tailers/file/tailer.go

```diff
@@ -25,26 +25,62 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/logs/tag"
 )
 
-// Tailer tails one file and sends messages to an output channel
+// Tailer tails a file, decodes the messages it contains, and passes them to a
+// supplied output channel for further processing.
+//
+// Operational Overview
+//
+// Tailers have three components, organized as a pipeline.  The first,
+// readForever, polls the file, trying to read more data.  That data is passed
+// to the second component, the decoder.  The decoder produces
+// decoder.Messages, which are passed to the third component, forwardMessages.
+// This component translates the decoder.Messages into message.Messages and
+// sends them to the tailer's output channel.
 type Tailer struct {
-	readOffset    int64
+	// lastReadOffset is the last file offset that was read.  This value must be
+	// accessed atomically.
```

Comment:
Would a mention of `get-` and `setLastReadOffset` make sense here?

Reply:
good point, yeah